src/main/java/core/basesyntax/validator/DataValidator.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax.validator;
+
+import core.basesyntax.model.Operation;
+import java.util.List;
+
+public class DataValidator {
+    private static final String COMMA = ",";
+
+    public void validate(List<String> inputData) {
+        if (inputData.isEmpty()) {
+            throw new IllegalArgumentException("Input data cannot be empty");
+        }
+
+        for (String line : inputData.subList(1, inputData.size())) {
+            String[] partsOfData = line.split(COMMA);
+            if (partsOfData.length != 3) {
+                throw new IllegalArgumentException("Incorrect input data in line: " + line);
```

Comment:
Consider providing more context in the error message or logging the error to help with debugging when the input data format is incorrect.